sycl/source/detail/queue_impl.hpp

```diff
@@ -825,11 +825,16 @@ class queue_impl {
       bool KernelUsesAssert = false;
 
       if (IsKernel)
-        // Kernel only uses assert if it's non interop one
+      // Kernel only uses assert if it's non interop one
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+        KernelUsesAssert = !(Handler.MKernel && Handler.MKernel->isInterop()) &&
+                           ProgramManager::getInstance().kernelUsesAssert(
+                               Handler.MKernelName.c_str());
```

Comment:
What would happen if we leave only this branch unconditionally?

Reply:
Are you asking if I can move this code before the condition` if (IsKernel)`?
According to the comment just above the line, "only kernels use assert".
So, I doubt it is correct to move it out of the condition.
Correct me if I am wrong.

Reply:
No, I'm asking what would happen if you change `#else` branch to be the same as this one.

Reply:
Aha!  
Yes, that will work since `const char*` can be implicitly converted to `std::string`.
I will change that.
Thanks.